### PR TITLE
RA 1117 Fix AddDiagnosisToVisitTest

### PR DESCRIPTION
**Pull request by cardy31**
- Test was unstable, so this test creates a new patient to make the
execution environment more stable
- Constructors fixed to no longer take driver as a parameter
- Cleaned up code style in a few places, i.e. one line functions
- Note: This pull request contains many, if not all of the same changes
found in request #116. If #116 is merged, the number of changed found
in this request should decrease significantly. However, I believe there
is enough new code solving a new problem to merit a standalone pull
request. Please correct me if I am wrong.
- Removed duplicate methods from VisitNotePage

### CODE REVIEW

File: ui-tests/src/main/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java [hunk 1]
```diff
@@ -290,13 +312,13 @@ public void clickChangeDate (){
 		clickOn(CHANGE_DATE);
 	}
 
-	public void enterDate(){
+	public void enterDate() {
 		Calendar currentDay = Calendar.getInstance();
 	 	Integer dayOfMoth = currentDay.get(Calendar.DAY_OF_MONTH);
-	 	A: for(int j = 6; j > 0; j--) {
+	 	A: for(int j = 6; j > 0; j--)
```
Comment (wluyima) [inline ui-tests/src/main/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java]: Our convention is to have curly braces around statements inside if clauses, can you please add them back?

File: ui-tests/src/main/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java [hunk 2]
```diff
@@ -290,13 +312,13 @@ public void clickChangeDate (){
...
-	public void enterDate(){
+	public void enterDate() {
 		Calendar currentDay = Calendar.getInstance();
 	 	Integer dayOfMoth = currentDay.get(Calendar.DAY_OF_MONTH);
-	 	A: for(int j = 6; j > 0; j--) {
+	 	A: for(int j = 6; j > 0; j--) 
 			for (int i = 7; i > 0; i--) {
 				WebElement day = findElement(By.xpath(String.format("//table[@class=' table-condensed']/tbody/tr[%s]/td[%s]",j, i)));
-				if(!day.getAttribute("class").contains("disabled")) {
+				if(!day.getAttribute("class").contains("disabled"))
```
Comment (wluyima) [inline ui-tests/src/main/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java]: Can you please add the curly braces back?

File: ui-tests/src/test/java/org/openmrs/reference/AddDiagnosisToVisitNoteTest.java [hunk 3]
```diff
@@ -10,34 +10,56 @@
...
+    public void AddDiagnosisVisitNoteTest() throws InterruptedException {
+        activeVisits = homePage.goToActiveVisitsSearch();
+        patientDashboardPage = activeVisits.goToPatientDashboardOfLastActiveVisit();
         visitNotePage = patientDashboardPage.goToVisitNote();
         visitNotePage.enterDiagnosis("Pne");
         visitNotePage.enterSecondaryDiagnosis("Bleed");
         assertEquals("Pneumonia", visitNotePage.primaryDiagnosis());
         assertEquals("Bleeding", visitNotePage.secondaryDiagnosis());
         visitNotePage.save();
-        assertNotNull(patientDashboardPage.visitLink());
```
Comment (wluyima) [inline ui-tests/src/test/java/org/openmrs/reference/AddDiagnosisToVisitNoteTest.java]: Why did you remove this?

Comment (cardy31) [inline ui-tests/src/test/java/org/openmrs/reference/AddDiagnosisToVisitNoteTest.java]: I took it out when I was trying to track down an error. I've added it back in. Thanks for pointing it out!

File: ui-tests/src/main/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java (hunk 2 above)
Comment (cardy31) [inline ui-tests/src/main/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java]: Will do! I'll adhere to that convention in the future. Thanks for pointing it out.

**Comment by dkayiwa** [comment]
@cardy31 do you see how this is failing consistently? https://ci.openmrs.org/browse/REFAPP-OMODDISTRO-4562

**Comment by cardy31** [comment]
@dkayiwa I do, but I can't replicate it locally. Both my branch and master run locally and pass. Do you have any suggestions? I'm at a loss since I can't replicate the failure.

**Comment by dkayiwa** [comment]
@cardy31 i have reverted it here https://github.com/openmrs/openmrs-distro-referenceapplication/commit/d217657b71a35d96fba7547eab45cc10526a1726 because it has consistently failed from: https://ci.openmrs.org/browse/REFAPP-OMODDISTRO-4555
